sound/soc/sof/pcm.c

```diff
@@ -414,8 +464,15 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		snd_sof_pcm_platform_trigger(sdev, substream, cmd);
 
 	/* free PCM if reset_hw_params is set and the STOP IPC is successful */
-	if (!ret && reset_hw_params)
+	if (!ret && reset_hw_params) {
 		ret = sof_pcm_dsp_pcm_free(substream, sdev, spcm);
+		if (ret < 0)
+			return ret;
+
+		/* free widget list during suspend */
```

Comment:
I don't understand the comment and what suspend has to do with trigger?

Reply:
sorry that was incorrect. I meant to say SUSPEND trigger not suspend.